src/DAL/migration/storedProcedure.sql

```diff
@@ -0,0 +1,75 @@
+        ) q
+    ) WITH NO DATA;
+
+    -- insert intersecting polygon parts into a temp table
+    INSERT INTO selected_parts
+	SELECT ST_IsEmpty(geom_intersections) is_empty_geom, ST_NumGeometries(geom_intersections) > 1 is_multi, *
+    FROM (
+        SELECT "internalId", ST_Difference(geom, r.geom) geom_intersections
+        FROM "PolygonParts".parts 
+        WHERE ST_Intersects(geom, r.geom)
+    ) q;
+
+    -- insert multi polygons that are the result of intersection with the input polygon
+    INSERT INTO "PolygonParts".parts(geom, "recordId", "productId", "productName", "productVersion", "sourceDateStart", "sourceDateEnd", "minResolutionDeg", "maxResolutionDeg", "minResolutionMeter", "maxResolutionMeter", "minHorizontalAccuracyCE90", "maxHorizontalAccuracyCE90", sensors, region, classification, description, "imageName", "productType", "srsName")
+    SELECT "parts_geom" geom, "recordId", "productId", "productName", "productVersion", "sourceDateStart", "sourceDateEnd", "minResolutionDeg", "maxResolutionDeg", "minResolutionMeter", "maxResolutionMeter", "minHorizontalAccuracyCE90", "maxHorizontalAccuracyCE90", sensors, region, classification, description, "imageName", "productType", "srsName"
+    FROM (SELECT (ST_Dump(geom_intersections)).geom parts_geom, p.*
+        FROM selected_parts
+        JOIN "PolygonParts".parts p
+        ON selected_parts."internalId" = p."internalId" AND selected_parts.is_multi IS true) q;
+
+    -- update geometries of polygon parts intersecting with input polygon
+    UPDATE "PolygonParts".parts
+    SET geom = selected_parts.geom_intersections
+    FROM selected_parts
+    WHERE parts."internalId" = selected_parts."internalId" AND selected_parts.is_empty_geom IS false AND selected_parts.is_multi IS false;
+    
+    -- delete completely covered polygon parts by the input polygon
+    DELETE FROM "PolygonParts".parts
+    WHERE (parts."internalId", true) IN (SELECT "internalId", is_empty_geom FROM selected_parts) OR
+    (parts."internalId", true) IN (SELECT "internalId", is_multi FROM selected_parts);
+    
+    -- insert the imput record
+    INSERT INTO "PolygonParts".parts("recordId", "productId", "productName", "productVersion", "sourceDateStart", "sourceDateEnd", "minResolutionDeg", "maxResolutionDeg", "minResolutionMeter", "maxResolutionMeter", "minHorizontalAccuracyCE90", "maxHorizontalAccuracyCE90", sensors, region, classification, description, geom, "imageName", "productType", "srsName")
+    VALUES(r.*);
+
+    -- clear the temp table
+    TRUNCATE selected_parts;
+    
+    COMMIT;
+END;
+$$;
+
+-- Usage example: CALL insert_polygon_part(('795813b2-5c1d-466e-8f19-11c30d395fcd', 'productId', 'productName', 'productVersion', '2022-08-22 02:08:10', '2022-08-22 02:08:10', 0.0001, 0.0001, 0.3, 0.3, 2.5, 2.5, 'sensors', NULL, 'Unclassified', 'description', 'SRID=4326;POLYGON((-20 51,10 51,10 56,-20 56,-20 51))', 'imageName', '', NULL)::"PolygonParts".insert_polygon_part_record);
```

Comment:
Please add a line at the end of the file.

Reply:
done